homeassistant/components/roku/media_player.py

```diff
@@ -234,6 +255,16 @@ def source_list(self) -> list:
         """List of available input sources."""
         return ["Home"] + sorted(app.name for app in self.coordinator.data.apps)
 
+    @roku_exception_handler
+    async def async_launch(self, app_id: str, content_id: str, media_type: str) -> None:
+        """Launch application with content deeplink."""
+        params = {
+            "contentID": content_id,
```

Comment:
Would be better to encode this parameter using `urllib.parse.quote` as some services like Spotify uses colon on their deeplink ids. For example  `spotify:playlist:5xddIVAtLrZKtt4YGLM1SQ` that would force the HA user to encode manually it before adding to the UI.